syncctl.py

```diff
@@ -0,0 +1,199 @@
+        p = subprocess.run(base_args + [chart], capture_output=True)
+        if p.returncode != 0:
+            if not "library charts are not installable" in str(p.stderr):
+                raise Exception(f'Error templating chart: {chart}, error: {p.stderr}')
+        yield p.stdout
+
+def extract_images(k8s_manifests: str) -> list[str]:
+    images = []
+    for d in yaml.load_all(k8s_manifests, Loader=yaml.SafeLoader):
+        # TODO: Handle kind: list + cephVersion
+        if d is None:
+            continue
+        if d["kind"] == "CronJob":
+            d["spec"] = d["spec"]["jobTemplate"]["spec"]
+        if d["kind"] in ["Deployment", "ReplicaSet", "StatefulSet", "DaemonSet", "Job", "CronJob", "ReplicationController"]:
+            containers = d["spec"]["template"]["spec"]["containers"]
+            if "initContainers" in d["spec"]["template"]["spec"]:
+                containers += d["spec"]["template"]["spec"]["initContainers"]
+            images += [container["image"] for container in containers]
+
+    # Special case Docker Hub
+    # https://github.com/containers/image/blob/1895e312af410ccdee5efa44e5223ec93ae76001/docs/containers-transports.5.md#dockerdocker-reference
+    for i, v in enumerate(images):
+        if "/" not in v:
+            images[i] = "docker.io/library/" + v
+        else:
+            s = v.split("/")
+            if not ("." in s[0] or ":" in s[0] or "localhost" == s[0]):
+                images[i] = "docker.io/" + v
+    return images
+
+def resolve_image(image: str) -> str:
+    p = subprocess.run(["skopeo", "inspect", f'docker://{image}'], capture_output=True)
+    if p.returncode != 0:
+        raise Exception(f'Error inspecting image: {image}, error: {p.stderr}')
+
+    return json.loads(p.stdout)["Digest"]
+
+def resolve_images(c: dict, manifest: dict) -> None:
+    images = {}
+    for m in template_charts(c["api_versions"], c["skip_charts"], c["values"]):
+        for image in extract_images(m):
+            if image not in images:
+                digest = resolve_image(image)
+                images[image] = {
+                    "registry": image.split("/")[0],
+                    "image": image[image.index("/")+1:min(image.index(":"), image.index("@"))],
+                    "digest": digest
+                }
+                if "@" not in image:
+                     images[image]["tag"] = image[image.index(":")+1:]
+    manifest["images"] = list(images.values())
+    save_manifest(manifest)
+
+def save_manifest(manifest: dict) -> dict:
+    with open("manifest.json.tmp", "w") as f:
+        json.dump(manifest, f, indent=4, sort_keys=True)
+    os.rename("manifest.json.tmp", "manifest.json")
+
+def load_manifest() -> dict:
+    with open("manifest.json") as f:
+        return json.load(f)
+
+def main() -> None:
+    args = parser.parse_args()
+
+    manifest = load_manifest()
+    os.makedirs("work", exist_ok=True)
+
+    if args.verbose:
+        basicConfig(level=DEBUG)
```

Comment:
```suggestion
        basicConfig(level=logging.DEBUG)
```
DEBUG is not defined, You need to import logging and use the package.